### PR TITLE
desed: 1.2.1 -> 1.2.2

### PR DESCRIPTION
**Pull request by vinylen**
https://github.com/SoptikHa2/desed/releases/tag/v1.2.2

Bumped desed version from v1.2.1 to v1.2.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/de/desed/package.nix [hunk 1]
```diff
@@ -3,22 +3,21 @@
   rustPlatform,
   fetchFromGitHub,
 }:
-
 rustPlatform.buildRustPackage {
```
Comment (NickCao) [inline pkgs/by-name/de/desed/package.nix:6]:
```suggestion
rustPlatform.buildRustPackage rec {
```

File: pkgs/by-name/de/desed/package.nix [hunk 2]
```diff
@@ -3,22 +3,21 @@
...
   pname = "desed";
-  version = "1.2.1-unstable-2024-09-06";
+  version = "1.2.2";
 
   src = fetchFromGitHub {
     owner = "SoptikHa2";
     repo = "desed";
-    rev = "master";
-    hash = "sha256-iCpEfefXXognk4YI1LLb3mwgaqMw4m3haq/gdS1JbQU=";
+    rev = "9dc58f95f49be8551001797f8d6065982f977fd7";
```
Comment (NickCao) [inline pkgs/by-name/de/desed/package.nix:13]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/de/desed/package.nix [hunk 3]
```diff
@@ -3,22 +3,21 @@
...
+    hash = "sha256-k844tZgZmT1Sx2GNf8nb8HBnMzS59GS+MgU37KoKQaU=";
   };
 
-  cargoHash = "sha256-z2qv394C0GhQC21HuLyvlNjrM65KFEZh1XLj+Y/B9ZM=";
+  cargoHash = "sha256-g9vc7FBiFRd9tS72YdL6IQ3dTYpyXnaND7MchO/Gfgg=";
 
   meta = {
-    changelog = "https://github.com/SoptikHa2/desed/releases/tag/v1.2.1";
+    changelog = "https://github.com/SoptikHa2/desed/releases/tag/v1.2.2";
```
Comment (NickCao) [inline pkgs/by-name/de/desed/package.nix:20]:
```suggestion
    changelog = "https://github.com/SoptikHa2/desed/releases/tag/v${version}";
```

**Comment by vinylen** [comment]
Hey @NickCao thanks for your feedback! I've updated the PR include all of your suggested changes.

**Comment by NickCao** [comment]
Oh but we are getting some hash mismatch:
```
error: hash mismatch in fixed-output derivation '/nix/store/k0k7fz3wzc0hhrsvxw754g9ngd1nr0zg-source.drv':
         specified: sha256-k844tZgZmT1Sx2GNf8nb8HBnMzS59GS+MgU37KoKQaU=
            got:    sha256-aKkOs8IhnHjoJkXq9ryGn9fN0AmZyVTHbD/Vano+Erw=
```

**Comment by a-kenji** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 348746`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>desed</li>
  </ul>
</details>

File: pkgs/by-name/de/desed/package.nix [hunk 4]
```diff
@@ -3,22 +3,21 @@
...
-  version = "1.2.1-unstable-2024-09-06";
+  version = "1.2.2";
 
   src = fetchFromGitHub {
     owner = "SoptikHa2";
     repo = "desed";
-    rev = "master";
-    hash = "sha256-iCpEfefXXognk4YI1LLb3mwgaqMw4m3haq/gdS1JbQU=";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-k844tZgZmT1Sx2GNf8nb8HBnMzS59GS+MgU37KoKQaU=";
```
Comment (a-kenji) [inline pkgs/by-name/de/desed/package.nix:14]:
```suggestion
    hash = "sha256-aKkOs8IhnHjoJkXq9ryGn9fN0AmZyVTHbD/Vano+Erw=";
```

File: pkgs/by-name/de/desed/package.nix [hunk 5]
```diff
@@ -3,22 +3,21 @@
...
   };
 
-  cargoHash = "sha256-z2qv394C0GhQC21HuLyvlNjrM65KFEZh1XLj+Y/B9ZM=";
+  cargoHash = "sha256-g9vc7FBiFRd9tS72YdL6IQ3dTYpyXnaND7MchO/Gfgg=";
```
Comment (a-kenji) [inline pkgs/by-name/de/desed/package.nix:17]:
```suggestion
  cargoHash = "sha256-3bkNirrvBOqIOhxD5xf3XsdaUMWFVT/lnN3gmZQIB9I=";
```

**Comment by vinylen** [comment]
This PR has been completely broken by me, i apologize in advance to anyone that has gotten a notification from this.

**Pull request closed without merging**